cli/src/config_handler.rs

```diff
@@ -0,0 +1,107 @@
+use anyhow::Result;
+use confy::{self, ConfyError};
+use serde::{Deserialize, Serialize};
+use std::{env, path::PathBuf};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MoonenvConfig {
+    pub default: Option<String>,
+
+    pub profiles: Vec<IndividualConfig>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct IndividualConfig {
+    pub url: Option<String>,
+
+    pub name: String,
+}
+
+impl Default for IndividualConfig {
+    fn default() -> Self {
+        Self {
+            name: "default".to_string(),
+            url: Some("www.moonenv.app".to_string()),
+        }
+    }
+}
+
+impl ::std::default::Default for MoonenvConfig {
+    fn default() -> Self {
+        Self {
+            default: Some("default".into()),
+            profiles: vec![IndividualConfig::default()],
+        }
+    }
+}
+
+fn get_config_path() -> PathBuf {
+    let home = env::var("HOME").expect("HOME environment variable not set");
```

Comment:
Perhaps here to "avoid" an exception, we could validate what is in the return of `echo ~`, considering that the developer is on MacOS or Linux. In theory, it will return the same as the one available in $HOME, if the developer has not defined it, it will not return an error.

I tried to find the equivalent for Windows, but I still haven’t found anything that works.

Reply:
Good catch. I don't have any Windows PC available yet; I found this [wiki](https://wiki.tcl-lang.org/page/HOMEPATH#:~:text=On%20a%20Microsoft%20Windows%20system,to%20the%20user's%20home%20directory.&text=That%20is%2C%20%25HOMEDRIVE%25%20is,JRandom%E2%80%9D%20(or%20whatever).). The variable name we should look for is `HOMEPATH`.
This [library](https://docs.rs/dirs/latest/dirs/fn.home_dir.html) that seems to handle it. It looks well maintained once the last push was 2 months ago, with 620 stars, few issues, etc... what do you think? Should we use it or handle it by ourselves?